src/profile-logic/call-tree.js

```diff
@@ -209,6 +214,41 @@ export class CallTree {
     };
   }
 
+  _getInliningBadge(
+    callNodeIndex: IndexIntoCallNodeTable,
+    funcName: string
+  ): ExtraBadgeInfo | void {
+    const calledFunction = getFunctionName(funcName);
+    const inlinedIntoNativeSymbol =
+      this._callNodeTable.sourceFramesInlinedIntoSymbol[callNodeIndex];
+    if (inlinedIntoNativeSymbol === null) {
+      return undefined;
+    }
+
+    if (inlinedIntoNativeSymbol === -1) {
+      return {
+        name: 'divergent-inlining',
+        vars: { calledFunction },
+        localizationId: 'CallTree--divergent-inlining-badge',
+        contentFallback: '',
+        titleFallback: `Some calls to ${calledFunction} were inlined by the compiler.`,
```

Comment:
Do you have an example for this case? I can't find one and I'm a bit curious 😄

Reply:
Great question, and a fun exercise for me!

I made a testcase where it happens: https://godbolt.org/z/Mecs6Tsj1

```cpp
int outer(int x) {
    // this call to inner is simple and gets inlined away
    int a = inner(x, false);

    // this call to inner runs a lot of code and is not inlined
    int b = inner(a, true);
    
    int c = a * b;
    return c;
}

int inner(int y, bool doComplicatedWork) {
    if (!doComplicatedWork) {
        return y * 3;
    }

    // Do something complicated that takes a lot of code.
    ...
}
```

```asm
outer(int):
        lea     r10d, [rdi+rdi*2] ; <-- this is the inlined y * 3 computation
        mov     edi, r10d
        call    inner(int, bool) [clone .part.0] ; <-- this is the non-inlined call
        imul    eax, r10d
        ret
```

Reply:
Another case where it could happen would be after call tree transformations. Say you have these two calls:

A -> B
A -> C -> B

And let's say the A -> B call gets inlined, but the A -> C call does not.

And now you add a "merge function" transform for C, so that you end up with two A -> B calls; one inlined and one not.

Reply:
Ah, interesting. Thanks for the explanation!